tests/specs/session-replay/mode.e2e.js

```diff
@@ -107,11 +160,54 @@ describe.withBrowsersMatching(notIE)('Session Replay Sample Mode Validation', ()
       mode: 1
     }))
 
-    await browser.execute(function () {
-      Object.values(NREUM.initializedAgents)[0].runtime.session.reset()
-    })
+    await Promise.all([
+      browser.execute(function () {
+        Object.values(NREUM.initializedAgents)[0].runtime.session.reset()
+      }), browser.pause(1000)
+    ])
+
+    await expect(getSR()).resolves.toEqual(expect.objectContaining({
+      recording: false,
+      initialized: true,
+      events: expect.any(Array),
+      mode: 0
+    }))
+  })
+
+  it('blocked => OFF => API does not restart', async () => {
+    await browser.url(await browser.testHandle.assetURL('rrweb-instrumented.html', config({ session_replay: { sampling_rate: 100, error_sampling_rate: 0 } })))
+      .then(() => browser.waitForSessionReplayRecording())
+
+    await expect(getSR()).resolves.toEqual(expect.objectContaining({
+      blocked: false,
+      recording: true,
+      initialized: true,
+      events: expect.any(Array),
+      mode: 1
+    }))
+
+    await Promise.all([
+      browser.execute(function () {
+        Object.values(NREUM.initializedAgents)[0].runtime.session.reset()
+      }), browser.pause(1000)
+    ])
+
+    await expect(getSR()).resolves.toEqual(expect.objectContaining({
+      blocked: true,
+      recording: false,
+      initialized: true,
+      events: expect.any(Array),
+      mode: 0
+    }))
+
+    await Promise.all([
+      browser.execute(function () {
+        newrelic.noticeError(new Error('test'))
+      }), browser.pause(1000)
+    ])
```

Comment:
Maybe add two more test cases:
1. Call the stop API in a script right below the loader. How does calling the api before the page load event affect the session replay state?
2. Call the start API in a script right below the loader. How does calling the api before the page load event affect the session replay state?